test/unit-test/ice_api/ice_api_utest.c

```diff
@@ -19,13 +102,337 @@
+    TEST_ASSERT_EQUAL( 10,
+                       context.maxLocalCandidates );
+    TEST_ASSERT_EQUAL( 0,
+                       context.numLocalCandidates );
+
+    TEST_ASSERT_EQUAL( &( remoteCandidateArray[ 0 ] ),
+                       context.pRemoteCandidates );
+    TEST_ASSERT_EQUAL( 10,
+                       context.maxRemoteCandidates );
+    TEST_ASSERT_EQUAL( 0,
+                       context.numRemoteCandidates );
+
+    TEST_ASSERT_EQUAL( &( candidatePairArray[ 0 ] ),
+                       context.pCandidatePairs );
+    TEST_ASSERT_EQUAL( 100,
+                       context.maxCandidatePairs );
+    TEST_ASSERT_EQUAL( 0,
+                       context.numCandidatePairs );
+
+    TEST_ASSERT_EQUAL( 1,
+                       context.isControlling );
+    TEST_ASSERT_EQUAL( &( transactionIdStore[ 0 ] ),
+                       context.pStunBindingRequestTransactionIdStore );
+    TEST_ASSERT_EQUAL( testRandomFxn,
+                       context.cryptoFunctions.randomFxn );
+    TEST_ASSERT_EQUAL( testCrc32Fxn,
+                       context.cryptoFunctions.crc32Fxn );
+    TEST_ASSERT_EQUAL( testHmacFxn,
+                       context.cryptoFunctions.hmacFxn );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate ICE Add Host Candidate fail functionality for Bad Parameters.
+ */
+void test_iceAddHostCandidate_BadParams( void )
+{
+    IceContext_t context;
+    IceEndpoint_t endPoint = { 0 };
+    IceResult_t result;
+
+    result = Ice_AddHostCandidate( NULL,
+                                   &( endPoint ) );
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_BAD_PARAM,
+                       result );
+
+    result = Ice_AddHostCandidate( &( context ),
+                                   NULL );
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_BAD_PARAM,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate ICE Add Host Candidate fail functionality for Max Candidate Threshold.
+ */
+void test_iceAddHostCandidate_MaxCandidateThreshold( void )
+{
+    IceContext_t context;
+    IceEndpoint_t endPoint = { 0 };
+    IceResult_t result;
+
+    context.numLocalCandidates = 1000;
+    context.maxLocalCandidates = 1000;
+
+    result = Ice_AddHostCandidate( &( context ),
+                                   &( endPoint ) );
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_MAX_CANDIDATE_THRESHOLD,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate ICE Add Host Candidate functionality.
+ */
+void test_iceAddHostCandidate( void )
+{
+    IceContext_t context = { 0 };
+    IceCandidate_t localCandidates[ 5 ];
+    IceEndpoint_t endPoint = { 0 };
+    IceResult_t result;
+
+    context.pLocalCandidates = &( localCandidates[ 0 ] );
+    context.maxLocalCandidates = 5;
+    context.numLocalCandidates = 0;
+
+    endPoint.isPointToPoint = 1;
```

Comment:
How about adding a test case with isPointToPoint = 0?

Reply:
Sure will do that as well.